frontend/ui/network/manager.lua

```diff
@@ -609,11 +607,31 @@ function NetworkMgr:reconnectOrShowNetworkMenu(complete_callback)
                 return
             end
         end
+        -- NOTE: Also supports a disconnect_callback, should we use it for something?
+        --       Tearing down Wi-Fi completely when tapping "disconnect" would feel a bit harsh, though...
+        UIManager:show(require("ui/widget/networksetting"):new{
+            network_list = network_list,
+            connect_callback = complete_callback,
+        })
+    end)
+end
+
+function NetworkMgr:reconnectOrShowNetworkMenu(complete_callback)
+    local info = InfoMessage:new{text = _("Scanning for networks…")}
+    UIManager:show(info)
+    UIManager:nextTick(function()
+        local network_list, err = self:getNetworkList()
+        UIManager:close(info)
+        if network_list == nil then
+            UIManager:show(InfoMessage:new{text = err})
+            return
+        end
         table.sort(network_list,
            function(l, r) return l.signal_quality > r.signal_quality end)
-
         local success = false
-        if not self.wifi_toggle_long_press then
+        if self.wifi_toggle_long_press then
+            self.wifi_toggle_long_press = false
```

Comment:
Possibly `nil`? (mostly for semantic/documentation concerns, makes it clearer that it's more about clearing a state variable that truly making the state false ;)).

Reply:
In which case it also wouldn't hurt to clear it unconditionally ;).